app/controllers/users_controller.rb

```diff
@@ -0,0 +1,83 @@
+class UsersController < ApplicationController
+  configure do
+    enable :sessions
+    set :session_secret, "secret"
+  end
+
+  get '/signup' do
+
+    if !logged_in?
+      erb :'users/create_user', locals: {message: "Please sign up before you sign in"}
+    else
+      redirect to '/tweets'
+    end
+  end
+
+  post '/signup' do
+
+    if params[:username] == "" || params[:email] == "" || params[:password] == ""
```

Comment:
I think you could use [ActiveRecord validations](http://guides.rubyonrails.org/active_record_validations.html) (e.g. `validates_presence_of :username`) to perform these types of checks more easily.